lib/tokens/IamTokenManager.js

```diff
@@ -32,6 +32,16 @@ class IAMTokenManager extends TokenManager {
 
     debug('Making IAM session request.');
     let accessToken;
+    let protocol;
```

Comment:
this agent creation block is pretty much identical to that in the `CookieTokenManager` - I suggest pulling the agent code up into the `TokenManager` so that it can be written/maintained in one place and used by both `CookieTokenManager` and `IamTokenManager`.

Reply:
During the refactoring I was also removed the agent reference from the Token Manager, because the new design does not need it anymore.
Thanks to this change the memory usage got cleaner surprisingly. I will soon update the PR description with the results.
The changes can be found here: 89d3fb0eb29f3b36000ddc34844c3eb8757cf640
Please let me know if there will be needed to make additional tests.

Reply:
What do you think about this aspect: a77b91f667fd